include/nuttx/spinlock.h

```diff
@@ -592,6 +592,7 @@ irqstate_t spin_lock_irqsave_wo_note(FAR volatile spinlock_t *lock)
       while (!spin_trylock_irqsave_wo_note(lock, ret));
     }
 
+  sched_lock_wo_note();
```

Comment:
should move to spin_trylock_wo_note, spin_lock without irq save sould disable sched too